src/common/System.cpp

```diff
@@ -410,19 +405,20 @@ void GenRandomBytes(void* dest, size_t size)
 	size_t bytes_written;
 	if (nacl_secure_random(dest, size, &bytes_written) != 0 || bytes_written != size)
 		Sys::Error("nacl_secure_random failed");
-#elif defined(__linux__) && defined(HAS_GETRANDOM_SYSCALL)
-	if (syscall(SYS_getrandom, dest, size, GRND_NONBLOCK) == -1)
-		Sys::Error("Failed getrandom syscall: %s", strerror(errno));
 #elif defined(__linux__)
-	int fd = open("/dev/urandom", O_RDONLY);
-	if (fd == -1)
-		Sys::Error("Failed to open /dev/urandom: %s", strerror(errno));
-	if (read(fd, dest, size) != (ssize_t) size)
-		Sys::Error("Failed to read from /dev/urandom: %s", strerror(errno));
-	close(fd);
+	if (getrandom(dest, size, GRND_NONBLOCK) == -1)
```

Comment:
It's nothing related to your change, but this `== -1` should be changed to `== size`. According to the man page it may return some number of bytes less than requested

Reply:
Right, this is fixed.

Also I noticed my code was stupid because if the random generator isn't ready yet, we would block on the fallback reading `urandom`, which completely defeats the purpose of using `getrandom` with `GRND_NONBLOCK`. So I rewrote the code to only use the fallback if `errno` is equal to `ENOSYS` (syscall not implemented).

Reply:
I also added a warning for when the syscall is not available, just before doing the fallback. All other errors are calling `Sys::Error`.

Reply:
Using nonblock never made any sense because we can't proceed without it. The flag can be removed

Reply:
Well maybe it makes sense if the idea is that we don't block indefinitely with no sign of what's happening (and instead print a warning before blocking). But if we keep nonblock I say we keep the fallback.